capstone/src/main/webapp/scripts/search.js

```diff
@@ -0,0 +1,76 @@
+const searchInput = document.getElementById("name-search");
+const suggestionsPanel = document.getElementById("suggestions");
+
+let names = [
+  { fullname: "Lucy Qu" },
+  { fullname: "Lucy Liu" },
+  { fullname: "Himani Yadav" },
+  { fullname: "Anika Bagga" },
+  { fullname: "Lucille Ball" }
+];
+let currentFocus = -1;
+
+function autocomplete() {
+  searchInput.addEventListener("keyup", filter);
+  searchInput.addEventListener("keyup", checkKey);
+}
+
+function filter() {
+  const input = searchInput.value.toLowerCase();
+  suggestionsPanel.innerHTML = "";
+  const suggestions = names.filter(function(name) {
+    return name.fullname.toLowerCase().startsWith(input);
+  });
+  suggestions.forEach(suggested => appendSuggestion(suggested));
+  if (input === "") {
+    suggestionsPanel.innerHTML = "";
+  }
+}
+
+function checkKey(e) {
+  let list = document.getElementById("suggestions");
+  if (list) {
+    list = list.getElementsByTagName("div");
+  }
```

Comment:
It is strange that you are reassigning the suggestions element to a list elements. Also, if "suggestions" does not exist, do you still want to pass that to the addActive function? I think you should split this into multiple variables and only continue with the key presses if "suggestions" exists

Reply:
edited in new pr